flow/device_flow.go

```diff
@@ -0,0 +1,200 @@
+// table, we need to evaluate if this makes sense. Should/Can we merge these two tables?
+//
+// DeviceFlowState* constants enumerate the states of a flow. The below graph
+// describes possible flow state transitions.
+//
+// graph TD
+//
+//	DEVICE_INITIALIZED --> DEVICE_UNUSED
+//	DEVICE_UNUSED --> DEVICE_USED
+//	DEVICE_UNUSED --> DEVICE_ERROR
+const (
+	// DeviceFlowStateLoginInitialized applies before the login app either
+	// accepts or rejects the login request.
+	DeviceFlowStateInitialized = int16(7)
+
+	// DeviceFlowStateUnused indicates that the login has been authenticated, but
+	// the User Agent hasn't picked up the result yet.
+	DeviceFlowStateUnused = int16(8)
+
+	// DeviceFlowStateUsed indicates that the User Agent is requesting consent and
+	// Hydra has invalidated the login request. This is a short-lived state
+	// because the transition to DeviceFlowStateConsentInitialized should happen while
+	// handling the request that triggered the transition to DeviceFlowStateUsed.
+	DeviceFlowStateUsed  = int16(9)
+	DeviceFlowStateError = int16(127)
+)
+
+type DeviceFlow struct {
+	ID  string    `db:"challenge"`
+	NID uuid.UUID `db:"nid"`
+
+	// RequestURL is the original OAuth 2.0 Device Authorization URL requested by the OAuth 2.0 client. This URL is typically not
+	// needed, but might come in handy if you want to deal with additional request parameters.
+	//
+	// required: true
+	RequestURL string `db:"request_url"`
+
+	// RequestedScope contains the OAuth 2.0 Scope requested by the OAuth 2.0 Client.
+	//
+	// required: true
+	RequestedScope sqlxx.StringSliceJSONFormat `db:"requested_scope"`
+
+	// RequestedAudience contains the access token audience as requested by the OAuth 2.0 Client.
+	//
+	// required: true
+	RequestedAudience sqlxx.StringSliceJSONFormat `db:"requested_at_audience"`
+
+	// Client is the OAuth 2.0 Client that initiated the request.
+	//
+	// required: true
+	Client *client.Client `db:"-"`
+
+	ClientID string `db:"client_id"`
+	Verifier string `db:"verifier"`
+	CSRF     string `db:"csrf"`
+
+	RequestedAt time.Time `db:"requested_at"`
+
+	State int16 `db:"state"`
+	// The user_code was already handled.
+	UserCodeWasUsed bool `db:"user_code_was_used"`
+	// HandledAt contains the timestamp the device user verification request was handled.
+	UserCodeHandledAt sqlxx.NullTime      `db:"user_code_handled_at"`
+	Error             *RequestDeniedError `db:"error"`
+}
+
+func NewDeviceFlow(r *DeviceUserAuthRequest) *DeviceFlow {
+	return &DeviceFlow{
+		ID:                r.ID,
+		Client:            r.Client,
+		ClientID:          r.ClientID,
+		RequestURL:        r.RequestURL,
+		Verifier:          r.Verifier,
+		CSRF:              r.CSRF,
+		RequestedAt:       r.RequestedAt,
+		UserCodeWasUsed:   r.WasHandled,
+		UserCodeHandledAt: r.HandledAt,
+		State:             DeviceFlowStateInitialized,
+	}
+}
+
+func (f *DeviceFlow) GetDeviceUserAuthRequest() *DeviceUserAuthRequest {
```

Comment:
```suggestion
func (f *DeviceFlow) NewDeviceUserAuthRequest() *DeviceUserAuthRequest {
```

why `Get` and not `New`?

Reply:
Follows the naming convention from flow (e.g. https://github.com/canonical/hydra/blob/canonical/flow/flow.go#L310)